lib/rb_shift/client.rb

```diff
@@ -62,11 +64,16 @@ def projects(update = false)
     end
 
     def execute(command, **opts)
-      `oc --server="#{@url}" --token="#{@token}" #{command}  #{unfold_opts opts}`
+      command = "oc --server=\"#{@url}\" --token=\"#{@token}\" #{command}  #{unfold_opts opts}"
+      Open3.popen3 command do |_, _, stderr, wait_thrd|
+        err_code = wait_thrd.value
+        raise InvalidCommandError unless stderr.gets.nil? && err_code.success?
+      end
     end
 
     def wait_project_deletion(project_name, timeout = 1)
-      sleep timeout until projects(true).find { |v| v.name == project_name }.nil?
+      sleep timeout until get('projects').find { |v| v[:metadata][:name] == project_name }.nil?
```

Comment:
Detect and find are identical in terms of implementation and behaviour. Personally I'd rather use .nil? method to explicitly state the predicate condition instead of expecting implicit behaviour, so the code is more understandable. On the other hand it's not something severe to change, but it will have to be changed all over project to unify style.